### PR TITLE
InventoryLevel

### PR DESCRIPTION
**Pull request by lukecyca**
This is a re-do of PR #159 which I messed up after getting @ttyfky's review.

This addresses the outstanding issue that prevented PR #102 from being merged.

This also takes the place of PR #133 which can be closed once this is merged.

### CODE REVIEW

**Comment by evermile-mitko** [comment]
Hey @ttyfky , is there any chance that you can take a look at this PR? I came to the PR with the need to use the InventoryLevel functionality from the client and it will be a shame to fork the repo only to have these changes integrated.

And @lukecyca - thanks for you effort on this!

**Comment by oliver006** [comment]
Thanks for your PR!
I recently joined as a maintainer and will start with reviewing and merging PRs in this repo as soon as possible.
There's a bit of a backlog and I will hopefully get to this PR soon to get it merged.

In the meantime, can you rebase the PR please? Thanks!

File: inventory_level.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+// InventoryLevel represents a Shopify inventory level
+type InventoryLevel struct {
+	InventoryItemID   int64      `json:"inventory_item_id,omitempty"`
+	LocationID        int64      `json:"location_id,omitempty"`
+	Available         int        `json:"available"`
+	CreatedAt         *time.Time `json:"created_at,omitempty"`
+	UpdatedAt         *time.Time `json:"updated_at,omitempty"`
+	AdminGraphqlAPIID string     `json:"admin_graphql_api_id,omitempty"`
```
Comment (oliver006) [inline inventory_level.go:33]: AdminGraphqlAPIID -> AdminGraphqlApiId

File: inventory_level.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+// InventoryLevelServiceOp is the default implementation of the InventoryLevelService interface
+type InventoryLevelServiceOp struct {
+	client *Client
+}
+
+// InventoryLevel represents a Shopify inventory level
+type InventoryLevel struct {
+	InventoryItemID   int64      `json:"inventory_item_id,omitempty"`
+	LocationID        int64      `json:"location_id,omitempty"`
```
Comment (oliver006) [inline inventory_level.go:29]: LocationId

File: inventory_level.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+// InventoryLevelServiceOp is the default implementation of the InventoryLevelService interface
+type InventoryLevelServiceOp struct {
+	client *Client
+}
+
+// InventoryLevel represents a Shopify inventory level
+type InventoryLevel struct {
+	InventoryItemID   int64      `json:"inventory_item_id,omitempty"`
```
Comment (oliver006) [inline inventory_level.go:28]: InventoryItemId

File: inventory_level.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+// InventoryLevelsResource is used for handling multiple item responsees
+type InventoryLevelsResource struct {
+	InventoryLevels []InventoryLevel `json:"inventory_levels"`
+}
+
+// InventoryLevelListOptions is used for get list
+type InventoryLevelListOptions struct {
+	InventoryItemIDs []int64   `url:"inventory_item_ids,omitempty,comma"`
```
Comment (oliver006) [inline inventory_level.go:48]: Ids

File: inventory_level.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	LocationIDs      []int64   `url:"location_ids,omitempty,comma"`
```
Comment (oliver006) [inline inventory_level.go:49]: Ids

File: inventory_level.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+	Limit            int       `url:"limit,omitempty"`
+	UpdatedAtMin     time.Time `url:"updated_at_min,omitempty"`
+}
+
+// InventoryLevelAdjustOptions is used for Adjust inventory levels
+type InventoryLevelAdjustOptions struct {
+	InventoryItemID int64 `json:"inventory_item_id"`
```
Comment (oliver006) [inline inventory_level.go:56]: Same

File: inventory_level.go [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+func (s *InventoryLevelServiceOp) List(options interface{}) ([]InventoryLevel, error) {
+	path := fmt.Sprintf("%s.json", inventoryLevelsBasePath)
+	resource := new(InventoryLevelsResource)
+	err := s.client.Get(path, resource, options)
+	return resource.InventoryLevels, err
+}
+
+// Delete an inventory level
+func (s *InventoryLevelServiceOp) Delete(itemID, locationID int64) error {
+	path := fmt.Sprintf("%s.json?inventory_item_id=%v&location_id=%v",
```
Comment (oliver006) [inline inventory_level.go:71]: `%d` for ints in the format string?

**Comment by oliver006** [comment]
@lukecyca - quick bump to see if you want to finish this PR. would love to get this merged but totally understand if you don't want to take this up again almost 2 years later.

**Review by oliver006 — APPROVED** [review]

**Pull request merged**